next/components/Molecules/Slider.tsx

```diff
@@ -0,0 +1,184 @@
+import { AnimatePresence, motion, PanInfo, Variant } from 'framer-motion'
+import { useTranslation } from 'next-i18next'
+import { wrap } from 'popmotion'
+import { FC, forwardRef, KeyboardEvent, ReactNode, useCallback, useEffect, useState } from 'react'
+
+type SliderProps = {
+  pages: ReactNode[]
+  autoSwipeDuration?: number
+  pagination?: FC<{
+    count: number
+    activeIndex: number
+    goToPage: (index: number) => void
+    goToPrevious: () => void
+    goToNext: () => void
+  }>
+  initialPage?: number
+  allowKeyboardNavigation?: boolean
+  description?: string
+}
+
+const variants: { [name: string]: Variant } = {
+  enter: (direction: number) => {
+    return {
+      x: direction > 0 ? 1000 : -1000,
+      opacity: 0,
+    }
+  },
+  center: {
+    zIndex: 1,
+    x: 0,
+    opacity: 1,
+  },
+  exit: (direction: number) => {
+    return {
+      zIndex: 0,
+      x: direction < 0 ? 1000 : -1000,
+      opacity: 0,
+    }
+  },
+}
+
+const swipeConfidenceThreshold = 10_000
+
+const swipePower = (offset: number, velocity: number) => {
+  return Math.abs(offset) * velocity
+}
+
+const Slider = forwardRef<HTMLDivElement, SliderProps>(
+  (
+    {
+      pages,
+      autoSwipeDuration = 0,
+      pagination: Pagination,
+      initialPage,
+      allowKeyboardNavigation = false,
+      description,
+    },
+    forwardedRef
+    // eslint-disable-next-line sonarjs/cognitive-complexity
+  ) => {
+    const { t } = useTranslation('common')
+    const [isFocused, setFocused] = useState(false)
+
+    const [[page, direction], setPage] = useState([initialPage ?? 0, 0])
+    const [isDragging, setDragging] = useState(false)
+    const index = wrap(0, pages.length, page)
```

Comment:
Wouldn't this work with simple modulus? e.g `mod` from lodash lib. (I know it's used in Marianum, but maybe we don't need another lib for it :D )